staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy.go

```diff
@@ -222,3 +228,97 @@ func MatchCustomResourceDefinition(label labels.Selector, field fields.Selector)
 func CustomResourceDefinitionToSelectableFields(obj *apiextensions.CustomResourceDefinition) fields.Set {
 	return generic.ObjectMetaFieldsSet(&obj.ObjectMeta, true)
 }
+
+// dropDisabledFields drops disabled fields that are not used if their associated feature gates
+// are not enabled.
+func dropDisabledFields(newCRD *apiextensions.CustomResourceDefinition, oldCRD *apiextensions.CustomResourceDefinition) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CustomResourceValidationExpressions) && !xValidationsInUse(oldCRD) {
+		if newCRD.Spec.Validation != nil {
+			dropXValidationsField(newCRD.Spec.Validation.OpenAPIV3Schema)
+		}
+		for _, v := range newCRD.Spec.Versions {
+			if v.Schema != nil {
+				dropXValidationsField(v.Schema.OpenAPIV3Schema)
+			}
+		}
+	}
+}
+
+// dropXValidationsField drops field XValidations from CRD schema
+func dropXValidationsField(schema *apiextensions.JSONSchemaProps) {
+	if schema == nil {
+		return
+	}
+	schema.XValidations = nil
+	if schema.AdditionalProperties != nil {
+		dropXValidationsField(schema.AdditionalProperties.Schema)
+	}
+	for def, jsonSchema := range schema.Properties {
+		dropXValidationsField(&jsonSchema)
+		schema.Properties[def] = jsonSchema
+	}
+	if schema.Items != nil {
```

Comment:
just out of curiosity, do we check the cases where the declared schema is not matched by existing persisted CR data?

Reply:
Basically it drops the field if feature gate is disabled when creation and FG is disabled + no persisted data with the field for update.

Reply:
We check the existing persisted CRD in PrepareForUpdate func above. For CR, we have similar process to check existing data while updating.